BOJ/1546/hikang.py

```diff
@@ -0,0 +1,12 @@
+# 시험 본 과목 개수 N개, 해당 시험 성적
+N = int(input())
+grade = list(map(int, input().split()))
+
+M = max(grade)
+sum = 0
+for i in range(N):
+  # 새로운 평균을 구하는 식
+  grade[i] = grade[i]/M*100
```

Comment:
```suggestion
  grade[i] = grade[i]*100/M
```

이렇게 하면 속도가 더 빨라집니다.
왜 그럴까~용?

Reply:
잘 모르겠습니다 선생님

Reply:
선생님....help~ヾ(＾∇＾)

Reply:
python이 내부적으로 지원해주는 type casting 때문이에요~

c언어에서는 실수형 자료를 다루려면 실수형 변수 선언을 해주거나 실수형 자료로 type casting을 해줘야 됩니다.
그런데 파이썬은 이 모든걸 자동으로 해줘서 편리하죠. 
그래서 `/`연산이 나오는 순간 `grade[i]`에 있는 변수는 정수형에서 실수형으로 바뀝니다.
그 다음에 `*100`을 해주는건 실수형 연산이 들어가는 거에요.

반면 `*100` 이후에 `/` 연산이 들어가면 실수형 연산은 한번밖에 없게 되어요.
`*100`을 해도 여전히 정수형 변수니까요.

다들 알겠지만 컴퓨터에게 연산을 시킬 때 `1.0 * 100 == 100.0` 이 나오는 것과 `1 * 100 == 100` 이 나오는 것의 차이에요.

시스템 프로그래밍 시간에 배워서 다들 알겠지만 정수형 연산에 비해 실수형 연산은 3배에서 5배까지 느린 연산이에요.
그래서 파이썬에서는 순서를 맞춰주는 것 만으로도 3~5배의 성능 향상을 할 수 있어요.

Reply:
오오~~~~~진짜 친절한 설명이다 담부터 순서 신경써서해볼게~!!!!!고마워